ui/components/app/toast-master/toast-master.js

```diff
@@ -0,0 +1,260 @@
+/* eslint-disable react/prop-types -- TODO: upgrade to TypeScript */
+
+import { isEvmAccountType } from '@metamask/keyring-api';
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { MILLISECOND, SECOND } from '../../../../shared/constants/time';
+import {
+  PRIVACY_POLICY_LINK,
+  SURVEY_LINK,
+} from '../../../../shared/lib/ui-utils';
+import { getAlertEnabledness } from '../../../ducks/metamask/metamask';
+import {
+  BorderColor,
+  BorderRadius,
+  IconColor,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import {
+  DEFAULT_ROUTE,
+  REVIEW_PERMISSIONS,
+} from '../../../helpers/constants/routes';
+import { getURLHost } from '../../../helpers/utils/util';
+import { getShowAutoNetworkSwitchTest } from '../../../pages/routes/routes-helpers';
+import {
+  getPermittedAccountsForCurrentTab,
+  getSelectedAccount,
+  getUseNftDetection,
+} from '../../../selectors';
+import { hidePermittedNetworkToast } from '../../../store/actions';
+import {
+  AvatarAccount,
+  AvatarAccountSize,
+  AvatarNetwork,
+  Icon,
+  IconName,
+} from '../../component-library';
+import { Toast, ToastContainer } from '../../multichain';
+import { SurveyToast } from '../../ui/survey-toast';
+import {
+  getNftDetectionEnablementToast,
+  getShowPrivacyPolicyToast,
+  getShowSurveyToast,
+  setNewPrivacyPolicyToastClickedOrClosed,
+  setNewPrivacyPolicyToastShownDate,
+  setShowNftDetectionEnablementToast,
+} from './toast-master-selectors';
+
+// Allow comparison with a previous value, in order to detect changes
+// (This pattern only works if ToastMaster is a singleton)
+let prevAccountAddress;
+
+export function ToastMaster({ props, context }) {
```

Comment:
Rather than having a React prop called `props`, could we instead elevate all the nested properties to be the React props so it's more intuitive to use, such as:

`<ToastMaster activeTavOrigin=... />`

Reply:
You kind of just caught this in the middle of a refactor, I knew that ultimately ToastMaster would have zero props, look again.